File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public boolean isPrime (long a){

        if(a<2) return false;

        for(int i=2;i<=Math.sqrt(a);i++){
            if(a%i==0){
                return false;
            }
        }
        return true;
    }

    public int solution(int n, int k) {
        String jin = Integer.toString(n,k);
        int answer= 0;
        StringTokenizer st = new StringTokenizer(jin,"0");
        while(st.hasMoreTokens()){
            long num = Long.parseLong(st.nextToken());
            if (isPrime(num))
                answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPrime(number){
    if(number==2) 
        return true
    if(number<2) 
        return false
    if(number%2==0)
        return false
    const sqrt=Math.sqrt(number)
    for(let i=3;i<=sqrt;i+=2)
        if(number%i===0)
            return false
    return true
}
function solution(n, k) {
    const transed=n.toString(k)
    const nums=transed.split('0').filter((elem)=>elem!=='').map(Number)
    return nums.filter(isPrime).length
}